x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -421,6 +423,14 @@ export class AlertsClient {
   }
 
   public async update({ id, data }: UpdateOptions): Promise<PartialAlert> {
+    return await retryIfConflicts(
+      this.logger,
+      `alertsClient.update(${id})`,
+      async () => await this.updateWithOCC({ id, data })
```

Comment:
Not directly related from this PR but whenever 409 errors are encountered, an orphaned API key will exist because the update call within `updateAlert` (`await this.unsecuredSavedObjectsClient.update<RawAlert>`) isn't wrapped in a try catch to then invalidate the created API key. Might be worth a new issue to track as this will likely happen more with the increase of 409 errors.

The same within `updateApiKeyWithOCC` and `enableWithOCC`.

Reply:
ya, I was worried about these side-effects with the retries; I'll check to see how messy it is, but will prolly just open a tech debt issue for it, thx!

Reply:
I opened issue https://github.com/elastic/kibana/issues/77953 for this, thx!